### DISCUSSION TITLE
Is there a way to disable the caching mechanism completely?

### DISCUSSION THREAD

**Discussion by theogravity** [urql-graphql/urql#2947] [Q&A] [answered]
I have an app that does not require any caching at all for queries / mutations, as in it shouldn't be doing it at all. Is there a way to disable the mechanism entirely?

**Comment by JoviDeCroock** [comment, marked as answer]
```js
import { dedupExchange, fetchExchange, createclient } from '@urql/core'

const client = createClient({
  url: '',
  exchanges: [dedupExchange, fetchExchange]
})
```

urql is completely pluggable so not including the `cacheExchange` removes caching

**Reply by kitten** [reply]
Too add to this 🎉, you may want to have a read through the "Architecture" page of the docs: https://formidable.com/open-source/urql/docs/architecture/#the-exchanges

There we describe how we handle and return results for "operations", which are basically our way of splitting up each GraphQL operation into an instruction that goes through a series of "exchanges" to then be served with an "Operation Result"

Caching is just one exchange in the control flow. That's why you can remove any exchanges responsible for caching, and everything will still work; just without caching
